contracts/AMMWrapper.sol

```diff
@@ -376,6 +390,20 @@ contract AMMWrapper is IAMMWrapper, ReentrancyGuard, BaseLibEIP712, SignatureVal
             // other ERC20 tokens
             IERC20(_order.makerAssetAddr).safeTransfer(_order.receiverAddr, _settleAmount);
         }
+        // Collect fee
+        if (_feeAmount > 0) {
+            if (_internalTxData.toEth) {
+                // Transfer WETH directly if internal maker asset is WETH
+                if (!_isInternalAssetETH(_internalTxData.makerAssetInternalAddr)) {
+                    weth.transfer(feeCollector, _feeAmount);
+                } else {
+                    payable(feeCollector).transfer(_feeAmount);
+                }
```

Comment:
Do you think it is a good idea to align the way of collecting fee with settling for receiver? 

It seems that receiver will always get ETH no matter internal asset is ETH or WETH. Maybe we could do the same for fee collector for better fund management?

Reply:
This saves some gas (1 eth transfer instead of 2) for every trade (with WETH as internal asset) and is worth it in my opinion but I can change if we think otherwise.

Reply:
Yeah, I think it's a cost vs operating problem, and there is no absolutely right answer. 😂 

Align the logic would make code simpler and more consistent (could even extract an util function to share the transfer logic for both receiver and fee collector), and fee collector would always receive ETH, easier for him to use fee as gas to send other tx in the future.

Actually I'm fine with both implementations. 💪 Just raise a different point of view here. Maybe we could refer to @charlesjhongc opinions.

Reply:
From the operational perspective, buyback only deal with WETH currently so I think fee should stay wrapped when possible. It will cause few inconsistencies in code but save some gas for user/relayer every time so the trade-off is worthy to me.